copy-frameworks script phase fails on Buddybuild with 0.35.0

* carthage install method: [x] homebrew
* `which carthage`: /usr/local/bin/carthage
* `carthage version`: 0.35.0
* `xcodebuild -version`: Xcode 11.5

The copy-frameworks build script fails. Builds that passed with 0.34.0 fail with 0.35.0

**copy-frameworks Output**
```
replaceItem failed: Error Domain=NSCocoaErrorDomain Code=512 "The file “DJXMLParser.framework” couldn’t be saved in the folder “Frameworks”." UserInfo={NSFileOriginalItemLocationKey=file:///tmp/sandbox/5bd727ab52b8370001d76b5f/bbtest/Build/Products/Debug-watchsimulator/WSJ%20Watch%20Extension.appex/Frameworks/DJXMLParser.framework/, NSURL=file:///tmp/sandbox/5bd727ab52b8370001d76b5f/bbtest/Build/Products/Debug-watchsimulator/WSJ%20Watch%20Extension.appex/Frameworks/DJXMLParser.framework/, NSFileNewItemLocationKey=file:///var/folders/1x/kp10v4pn1f1c6408hkznt50r0000gn/T/carthage-lipo-1i8vWX/DJXMLParser.framework, NSUnderlyingError=0x7ff4c7185a80 {**Error Domain=NSPOSIXErrorDomain Code=18 "Cross-device link"** UserInfo={NSURL=file:///tmp/sandbox/5bd727ab52b8370001d76b5f/bbtest/Build/Products/Debug-watchsimulator/WSJ%20Watch%20Extension.appex/Frameworks/DJXMLParser.framework/}}}
```

Both source and destination  on /tmp and /var exist and are valid.

Reverting manually to carthage 0.34.0 on the Buddybuild machine prior to running the script resolves the issue.

@jdhealy 

7528
    ✗ replaceItem failed: Error Domain=NSCocoaErrorDomain Code=512 "The file “iCarousel.framework” couldn’t be saved in the folder “Frameworks”." UserInfo={NSFileOriginalItemLocationKey=file:///tmp/sandbox/57627f1f6d58ca01004420c4/bbbuild/Build/Products/AdHoc-iphoneos/UK-FC.app/Frameworks/iCarousel.framework/, NSURL=file:///tmp/sandbox/57627f1f6d58ca01004420c4/bbbuild/Build/Products/AdHoc-iphoneos/UK-FC.app/Frameworks/iCarousel.framework/, NSFileNewItemLocationKey=file:///var/folders/1x/kp10v4pn1f1c6408hkznt50r0000gn/T/carthage-lipo-dzcozK/iCarousel.framework, NSUnderlyingError=0x7fc1ca06c0d0 {Error Domain=NSPOSIXErrorDomain Code=18 "Cross-device link" UserInfo={NSURL=file:///tmp/sandbox/57627f1f6d58ca01004420c4/bbbuild/Build/Products/AdHoc-iphoneos/UK-FC.app/Frameworks/iCarousel.framework/}}}
7529
    ⚠ Skipping duplicate build file in Compile Sources build phase: TDA/RestAPI/Classes/Models/Contact Form/TDAContactFormType+CoreDataProperties.swift (in target 'UK-FC' from project 'TDA')
7530
    ** BUILD FAILED **
7531
    The following build commands failed:
7532
    	PhaseScriptExecution Carthage /tmp/sandbox/57627f1f6d58ca01004420c4/bbbuild/Build/Intermediates.noindex/TDA.build/AdHoc-iphoneos/UK-FC.build/Script-91EB4D96220C49DB007E04E5.sh
7533
    (1 failure)

--- a/Source/CarthageKit/Xcode.swift
+++ b/Source/CarthageKit/Xcode.swift
@@ -1066,7 +1066,20 @@ private func stripBinary(_ binaryURL: URL, keepingArchitectures: [String]) -> Si
   }
   
   let replace: (URL, URL) -> SignalProducer<(), CarthageError> = { original, modified in
-    return fileManager.replaceItem(at: original, withItemAt: modified)
+    do {
+      let originalVolumeNumber = try FileManager.default.attributesOfFileSystem(forPath: original.deletingLastPathComponent().path)[.systemNumber] as? Int
+      let modifiedVolumeNumber = try FileManager.default.attributesOfFileSystem(forPath: modified.deletingLastPathComponent().path)[.systemNumber] as? Int
+      if originalVolumeNumber == modifiedVolumeNumber {
+        return fileManager.replaceItem(at: original, withItemAt: modified)
+      } else {
+        if FileManager.default.fileExists(atPath: original.path) {
+          try FileManager.default.removeItem(at: original)
+        }
+        return fileManager.copyItem(modified, into: original).map { _ in () }
+      }
+    } catch {
+      return .init(error: .internalError(description: error.localizedDescription))
+    }
   }
   
   return createTempDir

